test/src/l2_test.cpp

```diff
@@ -117,13 +117,20 @@ TEST_F(l2_test, init) {
 
   // Test failures
   bm_l2_deinit();
+  // Reset init count after deinit
+  init_count = 0;
   EXPECT_NE(bm_l2_init(NULL, NULL, num_devices), BmOK);
   EXPECT_NE(bm_l2_init(NULL, cfg, 0), BmOK);
   bm_task_create_fake.return_val = BmENOMEM;
   EXPECT_NE(bm_l2_init(NULL, cfg, num_devices), BmOK);
+  bm_l2_deinit();
+  // Reset init count after deinit
+  init_count = 0;
```

Comment:
We needed to reset the init_count here to prevent the segfault on the ubuntu machine. I also added it after every deinit just to be safe.

Reply:
Good call, together we spam init_count = 0 ;P